Makefile

```diff
@@ -148,51 +134,20 @@ test: core
 
 ########################## MAKE/INSTALL RULES ##########################
 #
-all: dependencies core
+all: core
 
 core:
-	@command -v $(firstword $(MSBUILD)) >/dev/null || (echo "OpenRA requires the '$(MSBUILD)' tool provided by Mono >= 5.4."; exit 1)
-ifeq ($(WIN32), $(filter $(WIN32),true yes y on 1))
-	@$(MSBUILD) -t:build -p:Configuration="Release-x86"
-else
-	@$(MSBUILD) -t:build -p:Configuration=Release
+	@command -v $(firstword $(MSBUILD)) >/dev/null || (echo "OpenRA requires the '$(MSBUILD)' tool provided by Mono >= 5.18."; exit 1)
+	@$(MSBUILD) -t:Build -restore -p:Configuration=Release -p:TargetPlatform=$(TARGETPLATFORM)
+ifeq ($(TARGETPLATFORM), unix-generic)
+	@./configure-system-libraries.sh
 endif
 
 clean:
-	@ $(MSBUILD) -t:clean
 	@-$(RM_F) *.config
-	@-$(RM_F) *.exe *.dll *.dylib ./OpenRA*/*.dll *.pdb mods/**/*.dll mods/**/*.pdb *.resources
+	@-$(RM_F) *.exe *.dll *.dll.config *.so *.dylib ./OpenRA*/*.dll *.pdb mods/**/*.dll mods/**/*.pdb *.resources
 	@-$(RM_RF) ./*/bin ./*/obj
-	@-$(RM_RF) ./thirdparty/download
-
-distclean: clean
-
-cli-dependencies:
-	@./thirdparty/fetch-thirdparty-deps.sh
-	@ $(CP_R) thirdparty/download/*.dll .
-	@ $(CP_R) thirdparty/download/*.dll.config .
-	@ test -f OpenRA.Game/obj/project.assets.json || $(MSBUILD) -t:restore
-
-linux-dependencies: cli-dependencies linux-native-dependencies
-
-linux-native-dependencies:
-	@./thirdparty/configure-native-deps.sh
-
-windows-dependencies: cli-dependencies
-ifeq ($(WIN32), $(filter $(WIN32),true yes y on 1))
-	@./thirdparty/fetch-thirdparty-deps-windows.sh x86
-else
-	@./thirdparty/fetch-thirdparty-deps-windows.sh x64
-endif
-
-osx-dependencies: cli-dependencies
-	@./thirdparty/fetch-thirdparty-deps-osx.sh
-	@ $(CP_R) thirdparty/download/osx/*.dylib .
-	@ $(CP_R) thirdparty/download/osx/*.dll.config .
-
-dependencies: $(os-dependencies)
```

Comment:
If we leave `dependencies` in the `Makefile`, but turn it into a no-op, that would mean the Snap package wouldn't need to drop the `make dependencies` call immediately this code is merged.

Reply:
Would this be to simplify parallel branches (playtest vs release)?